spec/requests/providers/home_address/non_uk_home_addresses_controller_spec.rb

```diff
@@ -0,0 +1,138 @@
+        expect(unescaped_response_body).to include("Enter your client's overseas home address")
+      end
+    end
+  end
+
+  describe "PATCH /providers/applications/:legal_aid_application_id/home_address/non_uk_home_address" do
+    subject(:patch_request) do
+      patch(
+        providers_legal_aid_application_home_address_non_uk_home_address_path(legal_aid_application),
+        params: address_params.merge(submit_button),
+      )
+    end
+
+    let(:submit_button) { {} }
+
+    context "when the provider is not authenticated" do
+      before { patch_request }
+
+      it_behaves_like "a provider not authenticated"
+    end
+
+    context "when the provider is authenticated" do
+      before do
+        login_as provider
+      end
+
+      context "with a valid address" do
+        context "when linked_applications flag is disabled" do
+          before { Setting.update!(linked_applications: false) }
+
+          it "redirects successfully to the next step" do
+            patch_request
+            expect(response).to redirect_to(providers_legal_aid_application_proceedings_types_path)
+          end
+        end
+
+        context "when linking_applications flag is enabled" do
+          before { Setting.update!(linked_applications: true) }
+
+          it "redirects successfully to the next step" do
+            patch_request
+            expect(response).to redirect_to(providers_legal_aid_application_copy_case_invitation_path)
+          end
+        end
+
+        it "creates an address record" do
+          expect { patch_request }.to change { applicant.addresses.count }.by(1)
+          expect(address.address_line_one).to eq(address_params[:non_uk_home_address][:address_line_one])
+          expect(address.address_line_two).to eq(address_params[:non_uk_home_address][:address_line_two])
+          expect(address.city).to eq(address_params[:non_uk_home_address][:city])
+          expect(address.county).to eq(address_params[:non_uk_home_address][:county])
+          expect(address.postcode).to be_nil
+        end
+      end
+
+      context "with an invalid address" do
+        before { address_params[:non_uk_home_address].delete(:country) }
+
+        it "renders the form again if validation fails" do
+          patch_request
+          expect(unescaped_response_body).to include("Enter a country")
+        end
+      end
+
+      context "with an already existing address" do
+        before { create(:address, applicant:) }
+
+        it "does not create a new address record" do
+          expect { patch_request }.not_to change { applicant.addresses.count }
+        end
+
+        it "updates the current address" do
+          patch_request
+          expect(address.address_line_one).to eq(address_line_one)
+          expect(address.address_line_two).to eq(address_line_two)
+          expect(address.city).to eq(city)
+          expect(address.county).to eq(county)
+          expect(address.country).to eq(country)
```

Comment:
(Similar to above) suggested change, add test that the location is set to "home": expect(address.location).to eq("home")